zebra-network/src/peer/connection.rs

```diff
@@ -394,6 +444,32 @@ where
                     tx,
                     span,
                 }),
+            (AwaitingRequest, PushTransaction(transaction)) => {
+                // Since we're not waiting for further messages, we need to
+                // send a response before dropping tx.
+                let _ = tx.send(Ok(Response::Nil));
+                self.peer_tx
```

Comment:
whats the difference between `tx` and `peer_tx` here?

Reply:
https://github.com/ZcashFoundation/zebra/pull/1016/files/409a2289b57aa71cacbf19b17f6f5839890092e3#diff-238888844b1b8121c9450ba2070242beR375

`tx` is the oneshot used to send the response to the request, `peer_tx` is the outbound message sink.